### ISSUE TITLE
Necromancy

### ISSUE THREAD

**Issue by SleepySteel** [jeremygurr/dcssca#289]
Necromutation should be castable by ghouls. I know it's because undead can't shapeshift, but come on- if an octopus can become a skeleton, shouldn't a rotten man be able to become a skeleton? Also, permanent death channel? Please maybe?

**Comment by Sandman25DCSS** [comment]
Necromutation would be strictly bad for Ghoul IMHO. Main benefits of Necromutation is rTorment and no hunger but that does not make much sense for Gh in CA fork. No potions and chance to be paralyzed by Confusion just for +6 AC and Necromancy enhancer?

**Comment by G0rg0** [comment]
Well, that's actually a good reason to allow it. It's certainly not going to be overpowered.

**Comment by jeremygurr** [comment]
Yeah but allowing guaranteed bad choices is not any better than creating only one optimal way to handle something. I like choices to be available, any of which may be a good choice under the right circumstances. If it's the wrong choice every time, then it's taking up valuable complexity space without providing any benefit.

**Comment by SleepySteel** [comment]
I don't feel like 6 AC and a necromancy booster can ever be an objectively wrong choice- it's got give and take like any choice would, and it's a choice I'd definitely make.

**Comment by jeremygurr** [comment]
This really opens up the question of whether undead should be able to transform or not. I'm not sure I buy into the idea that they should universally be banned from mutations and form changing, but that's the way it's been in vanilla for a while. I'd rather not make an exception for just this case. If this truly does make sense, I'd rather address the bigger issue of whether transmutation should be allowed for undead. 

Making undead effectively immune to mutations, by converting them to rotting, as is currently the case in vanilla crawl, is a significant buff to being undead. Making them not be able to transmute is a significant nerf. I wouldn't be totally against removing both constraints. 

As I see it, the undead nature is interesting and distinct enough with rN+++, rTorment, rPois, lack of hunger costs, rC+. That's quite a list of qualities already. I'm not sure they really need the other two major restrictions to keep them interesting and distinct from non-undead species. And in return, undead characters would have more gameplay options (and threats).

**Comment by yrmvgh** [comment]
Malmutation is one of the bigger annoyances in this game (not as fatal as new players think, but still quite annoying) and I enjoy having the option to take a break from that annoyance, when I feel like it, by choosing undead, Zin, or Jiv.

Here's another way you could look at it: mutation protection applies to ALL ghouls. Even ones who can't cast a single spell. But being unable to change forms just means that same ghoul can't quaff Lig. It doesn't care that it can't cast Spider Form, because it doesn't even have the skill to cast Magic Dart. So this character would be nerfed hard if you "gave" transmutation permission to it while also taking away mut immunity.

An undead wizard simply casts other spells. If we assume a focused conjurer is roughly as effective as a focused transmuter, then it's not harming you to just focus on other schools.

Long story short, I do think it would be an oddity or flavor glitch for one kind of undead to morph into another kind of undead. Vampires can cast many Tmut spells, but specifically can't use necromutation. I think that makes sense.

But if you did permit undead to use Tmut, I think that's more of a sidegrade, equivalent(ish) to other casting options, and shouldn't come with such a huge nerf. I would instead imply that changing forms is difficult for them by making a list of tmut spells that are restricted (like vampire). Or throwing an additional -1 on undead Tmut apts, except ofc vampire which already has its own specific Tmut permissions and restrictions.

**Comment by SleepySteel** [comment]
I don't think mutation immunity would strictly be a downgrade for ghouls- it's true that malmutation is a gigantic pain in the ass, but I'd absolutely want to try mutation roulette for claws 3 on a ghoul.

**Comment by jeremygurr** [comment]
I'm okay with only removing the transmutation restriction for undead. It seems inconsistent to me that living players can transmute to undead, but undead can't transmute to living forms. The only value of a living form over undead anyway is supposedly hp regeneration. The problem there is that mummy and ghoul already have normal hp regen, so that restriction is absolutely meaningless. Why not let mummy walk around in spider form if they wanted? They would lose their undead qualities in living forms, just like living species gain undead qualities in undead forms. 

Oh I got it. If we allowed unrestricted transmutation for undead (I don't see any significant value in restricting it, and it adds complexity for the players), then while in a living form, a ghoul _could_ mutate, allowing players to play mutation roulette if they really wanted to. Why not? Yet the players who wanted protection from mutations could just stay in undead form, or transmute to another undead form (if that ever made sense, maybe we can make more undead forms).

**Comment by yrmvgh** [comment]
Could also put a level 1 to 3 "illusion of life" Trmt spell in that has a long duration and makes you mostly alive for most purposes. Like necromutation, but awful unless you really want to mut up. :wink: 

But making use of existing spells sounds great, and is cleaner and easier - I propose illusion of life not as an alternative, but as an additional option

**Comment by jeremygurr** [comment]
Done. undead can now transmute, and if in an "alive" form, they can also mutate.
